[INFRA] add citation.cff... and SPRING CLEANING !!!!!

Remi-Gau opened the pull request:
- [x] lint code with miss hit as much as possible (through datalad run) (meaning most changes ot .m files were done manually)
- [x] manual fix of non controversial changes to pacify miss hit
  - [x] function name renaming so it matches filename
  - [x] remove non ascii characters
- [x] fix some warnings and dead code in doc
- [x] lint docs/conf.py with black
- [x] modify miss hit config to make it NOT complain given the current status of the code base

and eventually...
...
...
...

- [x] add CITATION.cff (and a makefile to help validate it)

Remi-Gau committed:
update miss hit config

Remi-Gau committed:
[DATALAD RUNCMD] mh_style --fix && mh_metric --ci && mh_l...

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "mh_style --fix && mh_metric --ci && mh_lint",
 "exit": 1,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

Remi-Gau committed:
remove non ascii characters

Remi-Gau committed:
fix readability warning and relint

Remi-Gau committed:
update miss hit config to make it VERY tolerant

Remi-Gau committed:
fix function names not matching filenames

Remi-Gau committed:
change miss hit config to current status

Remi-Gau committed:
fix some warnings and dead code in docs

Remi-Gau committed:
[DATALAD RUNCMD] lint conf.py

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "black docs",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

Remi-Gau committed:
update version in doc

Remi-Gau commented on docs/.gitignore:
```diff
@@ -0,0 +1 @@
+_build/*
```
by the way: why are the local builds of the doc not ignored?

if we want to have a "local" version, a PDF may be better, no?

Remi-Gau replied:
I am adding this gitignore.

But I am asking permission to nuke the _build from the repo.

jsheunis replied:
Probably just me not being attentive... I agree that we can kick it out.

Remi-Gau replied:
done

Remi-Gau commented:
BTW the read the docs build are failing. Let me know if you want me to have a look into this.

Remi-Gau committed:
add CITATION.cff

Remi-Gau commented on CITATION.cff:
```diff
@@ -0,0 +1,70 @@
+contact:
+  - family-names: "Heunis"
+    given-names: "Stephan"
+    affiliation: "FIXME"
+    email: FIXME@FIXME.com
```
@jsheunis you and other contributors should probably update those FIXME fields

I grabbed the names from the all contributors file

jsheunis replied:
Thanks Remi!

Remi-Gau commented on Makefile:
```diff
@@ -0,0 +1,26 @@
+help: ## Show what this Makefile can do
+	@python -c "$$PRINT_HELP_PYSCRIPT" < $(MAKEFILE_LIST)
+
+# may require a "pip install cffconvert" to run
+validate_cff: CITATION.cff
+	cffconvert --validate
+
```
more as doc for developers in case they don't remember how to validate those files

Remi-Gau committed:
add clean recipe in doc makefile

Remi-Gau committed:
[DATALAD RUNCMD] remove build folder in doc

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "cd docs && make clean",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^
